### PR TITLE
Add FameEX API support

### PR DESCRIPTION
**Pull request by YuriyVorobyov96**
[Support for FameEX API](https://fameex-docs.github.io/docs/api/spot/en/#introduction)

It should be noted that some methods may cause problems because the documentation is not up to date:
- some methods no longer exist;
- there are no descriptions for many error codes;
- parameters, their types and their mandatory values are not valid;
- also some methods return an invalid data set.

A few problematic APIs:
- `v1/api/spot/spot/fills` duplicates orders, because of which their number is not consistent with reality, should use `v1/api/spot/orderlist`;
- `v1/api/spot/orderlist` pageno parameter does not exist, instead you should send pageNum, startTime and endTime are not mandatory, but pageSize and pageNum are mandatory, in addition pageNum, pageSize, side and state are numeric values, not strings as specified in the documentation;
- `v1/api/orders_pending` no longer exists.

### CODE REVIEW

File: trade/api/fameex_api.js [hunk 1]
```diff
@@ -0,0 +1,491 @@
...
+    },
+
+    /**
+     * Get order details
+     * https://fameex-docs.github.io/docs/api/spot/en/#get-order-details
+     * @param {String} symbol The name of the currency pair, such as "BTC-USDT"
+     * @param {String} orderId Order ID | Example: '10918742125338689536'
+     * @return {Promise<Object>}
+     */
+    async getOrderDetails(symbol, orderId) {
```
Comment (confleux) [inline trade/api/fameex_api.js:289]:
```suggestion
    async getOrder(symbol, orderId) {
```

File: trade/trader_fameex.js [hunk 2]
```diff
@@ -0,0 +1,856 @@
...
+      try {
+        if (order.code === successCode) {
+          return {
+            orderId: order.data.orderId,
+            tradesCount: undefined, // FameEX doesn't provide trades
+            price: +order.data.price,
+            side: order.data.side === orderSides.buy ? 'buy' : 'sell',
+            type: formatOrderType(order.data.orderType),
+            amount: +order.data.money,
+            volume: +order.data.money * +order.data.triggerPrice,
```
Comment (confleux) [inline trade/trader_fameex.js:407]: See comments above

File: trade/trader_fameex.js [hunk 3]
```diff
@@ -0,0 +1,856 @@
...
+            pairPlain: coinPair.pairPlain,
+            pairReadable: coinPair.pairReadable,
+            totalFeeInCoin2: +order.data.filledFee,
+            amountExecuted: +order.data.filledAmount,
+            volumeExecuted: +order.data.filledMoney,
+            timestamp: order.data.createTime,
+            updateTimestamp: order.data.updateTime,
```
Comment (confleux) [inline trade/trader_fameex.js:414]: `createTime: 1694176590`

Whilst we need in format: `1 694 176 590 000`

```suggestion
            timestamp: order.data.createTime * 1000,
            updateTimestamp: order.data.updateTime * 1000,
```

File: trade/trader_fameex.js [hunk 4]
```diff
@@ -0,0 +1,856 @@
...
+        order = await fameEXApiClient.getOrderDetails(coinPair.pairDash, orderId);
+      } catch (error) {
+        log.warn(`API request getOrderDetails(${paramString}) of ${utils.getModuleName(module.id)} module failed. ${error}`);
+        return undefined;
+      }
+
+      try {
+        if (order.code === successCode) {
+          return {
+            orderId: order.data.orderId,
```
Comment (confleux) [inline trade/trader_fameex.js:401]:
```suggestion
            orderId: order.data.orderId.toString(),
```

File: trade/trader_fameex.js [hunk 5]
```diff
@@ -0,0 +1,856 @@
...
+            amountExecuted: +order.data.filledAmount,
+            volumeExecuted: +order.data.filledMoney,
+            timestamp: order.data.createTime,
+            updateTimestamp: order.data.updateTime,
+            status: formatOrderStatus(order.data.state),
+          };
+        } else {
+          const errorMessage = order.fameexErrorInfo ?? 'No details';
+          log.log(`Unable to get order ${orderId} details ${pair} pair: ${errorMessage}.`);
+          return undefined;
```
Comment (confleux) [inline trade/trader_fameex.js:420]:
```suggestion
         log.log(`Unable to get order ${orderId} details: ${JSON.stringify(errorMessage)}. Returning unknown order status.`);

          return {
            orderId,
            status: 'unknown', // Order doesn't exist or Wrong orderId
          };
```

File: trade/api/fameex_api.js [hunk 6]
```diff
@@ -0,0 +1,491 @@
...
+        data.price = price;
+      }
+
+      return protectedRequest('post', `${versioning.v1}/api/spot/orders`, data);
+    },
+
+    /**
+     * Cancel order
+     * https://fameex-docs.github.io/docs/api/spot/en/#cancel-order
+     * @param {String} symbol For example, the name of the currency pair: "BTC-USDT"
```
Comment (confleux) [inline trade/api/fameex_api.js:365]:
```suggestion
     * @param {String} symbol In FameEX format as 'BTC-USDT'
```

File: trade/api/fameex_api.js [hunk 7]
```diff
@@ -0,0 +1,491 @@
...
+     * @param {String} orderId Order ID | Example: '10918742125338689536'
```
Comment (confleux) [inline trade/api/fameex_api.js:366]:
```suggestion
     * @param {String} orderId Example: '10918742125338689536'
```

File: trade/api/fameex_api.js [hunk 8]
```diff
@@ -0,0 +1,491 @@
...
+        data.orderId = orderId;
+      }
+
+      return protectedRequest('post', `${versioning.v1}/api/spot/fills`, data);
+    },
+
+    /**
+     * Create order
+     * https://fameex-docs.github.io/docs/api/spot/en/#new-order
+     * @param {String} symbol For example, the name of the currency pair: "BTC-USDT"
```
Comment (confleux) [inline trade/api/fameex_api.js:334]:
```suggestion
     * @param {String} symbol In FameEX format as 'BTC-USDT'
```

File: trade/api/fameex_api.js [hunk 9]
```diff
@@ -0,0 +1,491 @@
...
+     * @return {Promise<Array>}
+    */
+    ticker() {
+      return publicRequest('get', `/api${versioning.v2}/ticker/24hr`, {});
+    },
+
+    /**
+     * Get depth data
+     * https://fameex-docs.github.io/docs/api/spot/en/#full-depth-returned-for-a-given-market-pair
+     * @param {String} symbol A pair such as "BTC_USDT"
```
Comment (confleux) [inline trade/api/fameex_api.js:438]:
```suggestion
     * @param {String} symbol In FameEX format as 'BTC-USDT'
```

File: trade/api/fameex_api.js [hunk 10]
```diff
@@ -0,0 +1,491 @@
...
+        orderId,
+      };
+
+      return protectedRequest('post', `${versioning.v1}/api/spot/cancel_orders`, data);
+    },
+
+    /**
+     * Cancel all order for specific symbol
+     * https://coinstore-openapi.github.io/en/index.html#one-click-cancellation
+     * @param {String} symbol For example, the name of the currency pair: "BTC-USDT"
```
Comment (confleux) [inline trade/api/fameex_api.js:381]:
```suggestion
     * @param {String} symbol In FameEX format as 'BTC-USDT'
```

File: trade/trader_fameex.js [hunk 11]
```diff
@@ -0,0 +1,856 @@
...
+
+    /**
+     * Get deposit address for specific coin
+     * @param coin e.g. BTC
+     * @returns {Promise<[]|undefined>}
+     */
+    async getDepositAddress(coin) {
+      const paramString = `coin: ${coin}`;
+
+      const networks = (this.currencyInfo(coin))?.networks || [];
```
Comment (confleux) [inline trade/trader_fameex.js:762]:
```suggestion
     const coinInfo = this.currencyInfo(coin);

      if (!coinInfo) {
        log.warn(`Unable to get deposit address for ${coin} on FameEX. I don't have info about coin ${coin}.`);
        return undefined;
      }

      const networks = coinInfo.networks;

```

File: trade/trader_fameex.js [hunk 12]
```diff
@@ -0,0 +1,856 @@
...
+            result.push({
+              network: networks[idx],
+              address: address.data.address,
+            });
+          }
+        });
+
+        return result;
+      } catch (e) {
+        log.warn(`Error while processing getDepositAddress(${paramString}) request: ${e}`);
```
Comment (confleux) [inline trade/trader_fameex.js:788]:
```suggestion
      } catch (error) {
        log.warn(`Error while processing getDepositAddress(${paramString}) request: ${error}`);
```

File: trade/trader_fameex.js [hunk 13]
```diff
@@ -0,0 +1,856 @@
...
+
+      const networks = (this.currencyInfo(coin))?.networks || [];
+
+      let addresses;
+      try {
+        addresses = await Promise.all(networks.map(async (network) => {
+          return fameEXApiClient.getDepositAddress(coin, network);
+        }));
+      } catch (err) {
+        log.warn(`API request getDepositAddress(${paramString}) of ${utils.getModuleName(module.id)} module failed. ${err}`);
```
Comment (confleux) [inline trade/trader_fameex.js:770]: `err -> error`

File: trade/trader_fameex.js [hunk 14]
```diff
@@ -0,0 +1,856 @@
...
+        return result;
+      } catch (e) {
+        log.warn(`Error while processing getDepositAddress(${paramString}) request: ${e}`);
+        return undefined;
+      }
+    },
+
+    async getFees(coinOrPair) {
+      // Not available for FameEX
+    },
```
Comment (confleux) [inline trade/trader_fameex.js:795]:
```suggestion
```

File: trade/trader_fameex.js (hunk 13 above)
Comment (confleux) [inline trade/trader_fameex.js:770]:
```suggestion
      } catch (error) {
        log.warn(`API request getDepositAddress(${paramString}) of ${utils.getModuleName(module.id)} module failed. ${error}`);
```

File: trade/trader_fameex.js [hunk 15]
```diff
@@ -0,0 +1,882 @@
...
+
+    module.exports.gettingMarkets = true;
+
+    return new Promise((resolve) => {
+      fameEXApiClient.markets().then((markets) => {
+        try {
+          const result = markets.data.reduce((acc, market) => {
+            const pair = formatPairName(market.pair);
+
+            acc[pair.pairReadable] = {
```
[the remark this answers is not in the document]
Comment (YuriyVorobyov96) [reply trade/trader_fameex.js:163]: As I can see

[p2pb2b store pairReadable](https://github.com/Adamant-im/adamant-tradebot/blob/master/trade/trader_p2pb2b.js#L39)
[stalecube store pairReadable to](https://github.com/Adamant-im/adamant-tradebot/blob/master/trade/trader_stakecube.js#L47)
[also azbit](https://github.com/Adamant-im/adamant-tradebot/blob/master/trade/trader_azbit.js#L41)
[only coinstore uses a different format 'cause we can't get a readable format](https://github.com/Adamant-im/adamant-tradebot/blob/master/trade/trader_coinstore.js#L129)

File: trade/trader_fameex.js [hunk 16]
```diff
@@ -0,0 +1,882 @@
...
+
+    if (module.exports.gettingMarkets) return;
+    if (module.exports.exchangeMarkets) return module.exports.exchangeMarkets[pair ? formatPairName(pair).pairReadable : pair];
+
+    module.exports.gettingMarkets = true;
+
+    return new Promise((resolve) => {
+      fameEXApiClient.markets().then((markets) => {
+        try {
+          const result = markets.data.reduce((acc, market) => {
```
[the remark this answers is not in the document]
Comment (YuriyVorobyov96) [reply trade/trader_fameex.js]: As for me reduce more readable :)
But I'll change to forEach to keep project style

File: trade/trader_fameex.js [hunk 17]
```diff
@@ -0,0 +1,882 @@
...
+            const pair = formatPairName(market.pair);
```
[the remark this answers is not in the document]
Comment (YuriyVorobyov96) [reply trade/trader_fameex.js:163]: I can't find pairNames in project

image

File: trade/trader_fameex.js [hunk 18]
```diff
@@ -0,0 +1,856 @@
...
+                  comment: undefined,
+                  confirmations: undefined,
+                  withdrawalFee: undefined,
+                  minWithdraw: +currency.min_withdraw,
+                  maxWithdraw: +currency.max_withdraw,
+                  logoUrl: undefined,
+                  exchangeAddress: undefined,
+                  decimals: undefined,
+                  precision: undefined,
+                  networks: networksByCurrency.get(currency.name.toUpperCase()),
```
[the remark this answers is not in the document]
Comment (YuriyVorobyov96) [reply trade/trader_fameex.js:127]: Fix it

File: trade/trader_fameex.js [hunk 19]
```diff
@@ -0,0 +1,856 @@
...
+        fameEXApiClient.currencies(),
+        fameEXApiClient.currenciesWithNetwork(),
+      ])
+          .then(([currenciesData, currenciesWithNetworks]) => {
+            try {
+              const result = {};
+
+              const currencies = currenciesData.data;
+
+              const networksByCurrency = currenciesWithNetworks.data.list.reduce((acc, data) => {
```
[the remark this answers is not in the document]
Comment (YuriyVorobyov96) [reply trade/trader_fameex.js:93]: I can't understand your point here

Can you explain?

File: trade/trader_fameex.js (hunk 15 above)
[the remark this answers is not in the document]
Comment (confleux) [reply trade/trader_fameex.js]: Private version is more fresh and it's a practise used there.

File: trade/trader_fameex.js (hunk 17 above)
[the remark this answers is not in the document]
Comment (confleux) [reply trade/trader_fameex.js:174]: Private version is more fresh and it's a practise used there.

File: trade/trader_fameex.js [hunk 20]
```diff
@@ -0,0 +1,856 @@
...
+
+      try {
+        ticker = await fameEXApiClient.ticker();
+      } catch (error) {
+        log.warn(`API request getRates(${paramString}) of ${utils.getModuleName(module.id)} module failed. ${error}`);
+        return undefined;
+      }
+
+      try {
+        ticker = ticker.find((t) => t.trading_pairs === coinPair.pairDash);
```
[the remark this answers is not in the document]
Comment (YuriyVorobyov96) [reply trade/trader_fameex.js:660]: Fixed

File: trade/trader_fameex.js [hunk 21]
```diff
@@ -0,0 +1,856 @@
...
+        book = await fameEXApiClient.orderBook(coinPair.pairPlain);
+      } catch (error) {
+        log.warn(`API request getOrderBook(${paramString}) of ${utils.getModuleName(module.id)} module failed. ${error}`);
+        return undefined;
+      }
+
+      try {
+        const result = {};
+
+        result.asks = book.data.asks.map((ask) => ({
```
[the remark this answers is not in the document]
Comment (YuriyVorobyov96) [reply trade/trader_fameex.js:703]: Hm, it makes sense
FameEX API responses really strange

[the remark this answers is not in the document]
Comment (YuriyVorobyov96) [reply trade/trader_fameex.js:703]: fixed

File: trade/trader_fameex.js [hunk 22]
```diff
@@ -0,0 +1,856 @@
...
+
+      try {
+        trades = await fameEXApiClient.getTradesHistory(coinPair.pairDash);
+      } catch (error) {
+        log.warn(`API request getTradesHistory(${paramString}) of ${utils.getModuleName(module.id)} module failed. ${error}`);
+        return undefined;
+      }
+
+      try {
+        const result = trades.map((trade) => ({
```
[the remark this answers is not in the document]
Comment (YuriyVorobyov96) [reply trade/trader_fameex.js]: fixed

File: trade/trader_fameex.js [hunk 23]
```diff
@@ -0,0 +1,856 @@
...
+        balances = await fameEXApiClient.getBalances();
+      } catch (error) {
+        log.warn(`API request getBalances(${paramString}) of ${utils.getModuleName(module.id)} module failed. ${error}`);
+        return undefined;
+      }
+
+      const spotWallet = balances.data.filter((wallet) => wallet.walletType === 'spot')[0].list;
+
+      try {
+        const result = spotWallet.map((crypto) => ({
```
[the remark this answers is not in the document]
Comment (YuriyVorobyov96) [reply trade/trader_fameex.js]: fixed

File: trade/trader_fameex.js [hunk 24]
```diff
@@ -0,0 +1,856 @@
...
+        };
+      }
+
+      if (!limit && side === 'sell' && !coin1Amount) {
+        message = `Unable to place an order on ${exchangeName} exchange at Market sell. Base amount ${marketInfo.coin1} is not provided.`;
+        log.warn(message);
+        return {
+          message,
+        };
+      }
```
[the remark this answers is not in the document]
Comment (YuriyVorobyov96) [reply trade/trader_fameex.js]: fixed

File: trade/trader_fameex.js (hunk 15 above)
[the remark this answers is not in the document]
Comment (YuriyVorobyov96) [reply trade/trader_fameex.js]: fixed

File: trade/trader_fameex.js (hunk 17 above)
[the remark this answers is not in the document]
Comment (YuriyVorobyov96) [reply trade/trader_fameex.js]: fixed

File: trade/trader_fameex.js [hunk 25]
```diff
@@ -0,0 +1,856 @@
...
+          orderId: order.orderId,
+          symbol: pair.pairReadable,
+          symbolPlain: pair.pairPlain,
+          price: +order.price,
+          side: order.side === orderSides.buy ? 'buy' : 'sell',
+          type: formatOrderType(order.orderType),
+          timestamp: order.createTime,
+          amount: +order.money,
+          amountExecuted: +order.filledAmount,
+          amountLeft: +order.filledAmount - +order.money,
```
[the remark this answers is not in the document]
Comment (YuriyVorobyov96) [reply trade/trader_fameex.js:353]: Hm
There is an example of answer for sell:
```
{
        "symbol": "DOGE-USDT",
        "orderId": "11920186459401486336",
        "clientOid": "",
        "side": 2,
        "orderType": 1,
        "price": "0.5",
        "amount": "60",
        "money": "0",
        "filledAmount": "0",
        "filledMoney": "0",
        "filledFee": "0",
        "deductFee": "0",
        "feeCurrency": "usdt",
        "triggerPrice": "0",
        "triggerType": "",
        "triggerState": 0,
        "liquidationType": 0,
        "strategyId": "",
        "strategyType": 0,
        "strategyName": "",
        "state": 2,
        "accountType": "spot",
        "platform": "API",
        "detail": {},
        "cancelType": 0,
        "createTime": 1694192436,
        "updateTime": 1694192438
    },
```

There is an example of answer for buy:
```
{
        "symbol": "DOGE-USDT",
        "orderId": "11920188580322607104",
        "clientOid": "",
        "side": 1,
        "orderType": 1,
        "price": "0.5",
        "amount": "60",
        "money": "0",
        "filledAmount": "60",
        "filledMoney": "3.783",
        "filledFee": "0.06",
        "deductFee": "0.02997621",
        "feeCurrency": "doge",
        "triggerPrice": "0",
        "triggerType": "",
        "triggerState": 0,
        "liquidationType": 0,
        "strategyId": "",
        "strategyType": 0,
        "strategyName": "",
        "state": 4,
        "accountType": "spot",
        "platform": "API",
        "detail": {},
        "cancelType": 0,
        "createTime": 1694192941,
        "updateTime": 1694192943
    }
```

I can't understand
The right value is the `amount`, right? So what is `filledAmount`

File: trade/trader_fameex.js [hunk 26]
```diff
@@ -0,0 +1,856 @@
...
+        );
+      } catch (error) {
+        log.warn(`API request getOpenOrders(${paramString}) of ${utils.getModuleName(module.id)} module failed. ${error}`);
+        return undefined;
+      }
+
+      const orders = ordersData?.data?.orders || [];
+
+      try {
+        const result = orders.map((order) => ({
```
[the remark this answers is not in the document]
Comment (YuriyVorobyov96) [reply trade/trader_fameex.js]: fixed

File: trade/trader_fameex.js [hunk 27]
```diff
@@ -0,0 +1,856 @@
...
+        ordersInfo = await this.getOpenOrdersPage(coinPair, pageNum);
+
+        if (!ordersInfo) return undefined;
+
+        allOrders.push(...ordersInfo);
+
+        pageNum += 1;
+      } while (ordersInfo.length === ordersMaxPageSize);
+
+      return allOrders;
```
[the remark this answers is not in the document]
Comment (YuriyVorobyov96) [reply trade/trader_fameex.js]: Why here if we have it inner function?

File: trade/trader_fameex.js [hunk 28]
```diff
@@ -0,0 +1,856 @@
...
+          symbol: pair.pairReadable,
+          symbolPlain: pair.pairPlain,
+          price: +order.price,
+          side: order.side === orderSides.buy ? 'buy' : 'sell',
+          type: formatOrderType(order.orderType),
+          timestamp: order.createTime,
+          amount: +order.money,
+          amountExecuted: +order.filledAmount,
+          amountLeft: +order.filledAmount - +order.money,
+          status: formatOrderStatus(order.state),
```
[the remark this answers is not in the document]
Comment (YuriyVorobyov96) [reply trade/trader_fameex.js]: Why not? 
Isn't it better than reading a series of if/else constructs that pollute the code?
Plus this part of the code is repeated in two methods

File: trade/trader_fameex.js (hunk 11 above)
Comment (YuriyVorobyov96) [reply trade/trader_fameex.js]: fixed

File: trade/trader_fameex.js (hunk 13 above)
Comment (YuriyVorobyov96) [reply trade/trader_fameex.js]: fixed

File: trade/trader_fameex.js (hunk 25 above)
[the remark this answers is not in the document]
Comment (confleux) [reply trade/trader_fameex.js]: Lets say you place an order to buy 1 ETH for 1500 USDT and the lowest existing ask in order book is
```
{amount: 0.3, price: 1500}
```
Then, exchange will sell you 0.3 ETH for 1500 USDT (that's 450 USD)

And lets say next lowest price order in order book has price 2000 USDT for some time.

We get order data and get:
Amount: 1,
Executed amount: 0.3
Amount left: 0.7

[the remark this answers is not in the document]
Comment (YuriyVorobyov96) [reply trade/trader_fameex.js]: Fixed

File: trade/trader_fameex.js (hunk 18 above)
[the remark this answers is not in the document]
Comment (confleux) [reply trade/trader_fameex.js:127]:
```
ERC20: {
      id: '5',
      chainName: 'ERC20',
      status: true,
      withdrawalFee: 6,
      minWithdraw: NaN,
      confirmations: NaN
    },
```

File: trade/trader_fameex.js (hunk 19 above)
[the remark this answers is not in the document]
Comment (confleux) [reply trade/trader_fameex.js:93]: Ok, now I see it's legit.
Would be better with `for ... of`

File: trade/trader_fameex.js (hunk 18 above)
[the remark this answers is not in the document]
Comment (YuriyVorobyov96) [reply trade/trader_fameex.js:127]: fixed

File: trade/trader_fameex.js [hunk 29]
```diff
@@ -0,0 +1,856 @@
...
+      const orders = ordersData?.data?.orders || [];
+
+      try {
+        const result = orders.map((order) => ({
+          orderId: order.orderId,
+          symbol: pair.pairReadable,
+          symbolPlain: pair.pairPlain,
+          price: +order.price,
+          side: order.side === orderSides.buy ? 'buy' : 'sell',
+          type: formatOrderType(order.orderType),
```
[the remark this answers is not in the document]
Comment (YuriyVorobyov96) [reply trade/trader_fameex.js]: fixed

File: trade/trader_fameex.js [hunk 30]
```diff
@@ -0,0 +1,856 @@
...
+    },
+
+    /**
+     * List of all account open orders
+     * !POSSIBLE IMPLEMENTATION ERRORS!
+     * !At the moment it is impossible to implement this functional correctly, due to problems on the FameEX side
+     * @param {String} pair In classic format as BTC/USDT
+     * @returns {Promise<Array|undefined>}
+     */
+    async getOpenOrders(pair) {
```
[the remark this answers is not in the document]
Comment (YuriyVorobyov96) [reply trade/trader_fameex.js:375]: same as try/catch

what is the reason if we have inner function with this data?

File: trade/trader_fameex.js (hunk 3 above)
Comment (YuriyVorobyov96) [reply trade/trader_fameex.js]: fixed

File: trade/trader_fameex.js (hunk 19 above)
[the remark this answers is not in the document]
Comment (YuriyVorobyov96) [reply trade/trader_fameex.js]: fixed

**Review by confleux — APPROVED** [review]

File: trade/api/fameex_api.js (hunk 1 above)
Comment (confleux) [reply trade/api/fameex_api.js:286]: ?

File: trade/api/fameex_api.js [hunk 31]
```diff
@@ -0,0 +1,481 @@
...
+     * !POSSIBLE IMPLEMENTATION ERRORS!
+     * !At the moment it is impossible to implement this functional correctly, due to problems on the FameEX side
+     * @param {String} base Transaction currency (uppercase, such as "BTC")
+     * @param {String} quote Denominated currency (uppercase, such as "USDT")
+     * @param {Number} pageNum Pagination, the first few pages (1 <= pageNum)
+     * @param {Number} pageSize Pagination, the number of pages (1 < pageSize <= 500)
+     * @param {String} orderId Order ID
+     * @return {Promise<Object>}
+     */
+    async getTransactionDetails(
```
Comment (confleux) [inline trade/api/fameex_api.js:307]: Do we actually need this?

Comment (YuriyVorobyov96) [reply trade/api/fameex_api.js:307]: Hm, no, I already removed it from the Trader, but forgot to remove from API

Comment (YuriyVorobyov96) [reply trade/api/fameex_api.js]: fixed

File: trade/api/fameex_api.js (hunk 1 above)
Comment (YuriyVorobyov96) [reply trade/api/fameex_api.js]: fixed

**Review by adamant-al — APPROVED** [review]

**Pull request merged**